cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/FileSystemStorageAdapter.java

```diff
@@ -353,12 +355,14 @@ public StorageMetadata put(NewArtifact newArtifact, InputStream source)
             log.debug("calculated md5sum: " + checksum);
             log.debug("calculated file size: " + length);
             
+            IncorrectContentChecksumException checksumException = null;
+            boolean checksumProvided = newArtifact.contentChecksum != null && newArtifact.contentChecksum.getScheme().equals(MD5_CHECKSUM_SCHEME);
             // checksum comparison
-            if (newArtifact.contentChecksum != null && newArtifact.contentChecksum.getScheme().equals(MD5_CHECKSUM_SCHEME)) {
+            if (checksumProvided) {
                 String expectedMD5 = newArtifact.contentChecksum.getSchemeSpecificPart();
                 String actualMD5 = checksum.getSchemeSpecificPart();
                 if (!expectedMD5.equals(actualMD5)) {
-                    throw new StreamCorruptedException(
+                    checksumException = new IncorrectContentChecksumException(
```

Comment:
Don't use local var for the exception; just throw it immediately with "throw new ..."